nuxeo-importer-core/src/main/java/org/nuxeo/ecm/platform/importer/service/DefaultImporterServiceImpl.java

```diff
@@ -139,15 +139,16 @@ protected SourceNode createNewSourceNodeInstanceForSourcePath(String sourcePath)
     }
 
     protected ImporterDocumentModelFactory getDocumentModelFactory() {
-        if (documentModelFactory == null) {
-            if (docModelFactoryClass != null
-                    && DefaultDocumentModelFactory.class.isAssignableFrom(docModelFactoryClass)) {
-                try {
-                    setDocumentModelFactory(docModelFactoryClass.getConstructor(String.class, String.class).newInstance(
-                            getFolderishDocType(), getLeafDocType()));
-                } catch (ReflectiveOperationException e) {
-                    log.error(e, e);
-                }
+        if (documentModelFactory == null && docModelFactoryClass != null) {
+            try {
+            	if (DefaultDocumentModelFactory.class.isAssignableFrom(docModelFactoryClass)) {
+            		setDocumentModelFactory(docModelFactoryClass.getConstructor(String.class, String.class)
+                                                				.newInstance(getFolderishDocType(), getLeafDocType()));
+            	} else {
+            		setDocumentModelFactory(docModelFactoryClass.getConstructor().newInstance());
+            	}
+            } catch (ReflectiveOperationException e) {
+                log.error(e, e);
```

Comment:
There are tabs indents here, you should use spaces.. Also while refactoring please don't keep the `log.error` when catching an exception, just rethrow as a `new NuxeoException(e)`.
For clarity (no code repetition) assign the constructor to a single `ctor` variable in each branch of the if/else and after that block do the `setDocumentModelFactory(ctor)`.